libdnf5/conf/config_main.cpp

```diff
@@ -84,6 +87,15 @@ static int str_to_bytes(const std::string & str) {
     return static_cast<int>(res);
 }
 
+static std::string get_user_agent() {
+    utils::OSRelease os_release;
+    return std::format("libdnf ({} {}; {}; {}.{})",
+        os_release.get_value("NAME"),
+        os_release.get_value("VERSION_ID"),
+        os_release.get_value("VARIANT_ID"),
+        utils::get_os(),
+        rpm::get_base_arch(utils::detect_arch()));
```

Comment:
Old libdnf had a fallback if either `NAME`, `VERSION_ID`, `os` or `arch` were missing it set just `'libdnf'` user agent. Can we do that here as well to be compatible?